packages/core/src/map.ts

```diff
@@ -0,0 +1,69 @@
+export class Map<K, V> {
+    private map: { key: K; value: V }[] = []
```

Comment:
it would be faster and more memory efficient to have two arrays - keys and values and then instead of `findIndexByKey()` just do `keys.indexOf()`

Reply:
Amazing suggestion! Thank you